hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/ContainerEndpoint.java

```diff
@@ -237,14 +237,18 @@ public Response getReplicaHistoryForContainer(
    * {@link org.apache.hadoop.ozone.recon.api.types.MissingContainerMetadata}
    * for all missing containers.
    *
+   * @param limit The limit of missing containers to return.
    * @return {@link Response}
    */
   @GET
   @Path("/missing")
-  public Response getMissingContainers() {
+  public Response getMissingContainers(
+      @DefaultValue(DEFAULT_FETCH_COUNT) @QueryParam(RECON_QUERY_LIMIT)
+      int limit
+  ) {
     List<MissingContainerMetadata> missingContainers = new ArrayList<>();
     containerHealthSchemaManager.getUnhealthyContainers(
-        UnHealthyContainerStates.MISSING, 0, Integer.MAX_VALUE)
+        UnHealthyContainerStates.MISSING, 0, limit)
```

Comment:
Will the older or newer unhealthy containers show up in the result when we add limit here?

Reply:
It will go from older to limit in my belief. As the `getUnhealthyContainers` is creating a `SelectQuery` where the offset is 0 and the limit is set by the limit variable. So it adds a 0-based OFFSET clause to the query and it will limit the results of this select. The age of the unhealthy container counts from the `inStateSince` variable or when was it inserted into the DB via the `insertUnhealthyContainerRecords` method?

Reply:
I see. I have an idea where it could make sense to somehow append `ORDER BY time DESC` to the SQL query to get the newer unhealthy containers now that we are returning limited results. But I'm fine with not doing it in this jira if the change would be complicated.